File: benchmark/php/newton_sqrt.php
<?php
function sqrt_newton($n) {
    if ($n == 0) return 0;
    $x = $n / 2;
    for ($i = 0; $i < 20; $i++) { $x = ($x + $n / $x) / 2; }
    return $x;
}
$start = hrtime(true);
$sum = 0;
for ($i = 1; $i <= 10000; $i++) { $sum += sqrt_newton($i); }
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Sum: " . $sum . "
";
echo "Time: " . round($ms) . "ms
";

File: benchmark/perl/newton_sqrt.pl
use Time::HiRes qw(time);
sub sqrt_newton {
    my $n = shift; return 0 if $n == 0;
    my $x = $n / 2;
    for (1..20) { $x = ($x + $n / $x) / 2; }
    return $x;
}
my $start = time();
my $sum = 0;
for my $i (1..10000) { $sum += sqrt_newton($i); }
my $end = time();
my $ms = ($end - $start) * 1000;
print "Sum: $sum
";
printf "Time: %.0fms
", $ms;
